server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTreeBelowSGCachedImpl.java

```diff
@@ -670,6 +674,13 @@ protected void collectMeasurement(IMeasurementMNode node) {
               // only when user query with alias, the alias in path will be set
               path.setMeasurementAlias(node.getAlias());
             }
+            if (withTags) {
+              try {
+                path.setTagMap(tagManager.readTagFile(node.getOffset()).getLeft());
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
             result.add(path);
```

Comment:
I think the ```MTree``` shall not dependent on the ```TagManager```. What about passing a ```Function``` or ```Consumer```, the implementation of which is to load tag info into measurementPath, into this method to decouple ```MTree``` with ```TagManager```.

Reply:
Good catch. I have refined these codes as below:
1. Use `Function<IMeasurementNode, Map<String, String>>` as a tagGetter instead of calling `TagManager` directly.
2. Add a method `readTags` in `TagManager`.